hedera-node/src/main/java/com/hedera/services/txns/contract/operation/HederaCreateOperation.java

```diff
@@ -45,17 +47,24 @@ public Gas cost(final MessageFrame frame) {
 
 		//TODO charge for memory expansion cost?
 		//final Gas memoryGasCost = gasCalculator().memoryExpansionGasCost(frame, initCodeOffset, initCodeLength);
-
-		WorldStateAccount hederaAccount =
-				((HederaWorldUpdater) frame.getWorldUpdater()).getHederaAccount(frame.getRecipientAddress());
-		if (hederaAccount == null) {
-			hederaAccount = ((HederaWorldUpdater) frame.getWorldUpdater()).getHederaAccount(frame.getSenderAddress());
+		HashMap<Address, Long> map = frame.getMessageFrameStack().getLast().getContextVariable("expiries");
+		long expiry = 0;
+		if (!map.containsKey(frame.getSenderAddress())) {
```

Comment:
What about an indirect CREATE, do we want the expiry of the transaction sender (this) or the expiry of the contract doing the CREATE (indirect)?

Reply:
As far as I know, we need the expiry of the contract that is creating it.